backend/dataall/base/services/IdentityProvider.py

```diff
@@ -0,0 +1,8 @@
+import abc
```

Comment:
Here we are introducing a new directory in the `base` package called `services`. I see where the name comes from, we are defining "non-aws generic services". But since we are already using `services` quite a lot in the repo, when we refer to business logic of a particular data.all object, I think the name might lead to confusion.

Reply:
- Option 1: we rename the `services` directory
- Option 2: The package `aws` could be called something else like `cloud` and would account for all cloud-services. That way we could define IdentityProviders as part of the newly-named `aws` package. Since, right now IdentityProviders is only used for Cognito. I suggest that for the time being we move it to `aws`.
- Option 3: we move IdentityProviders to move to `utils`

Reply:
I like option 3. I will place the IdentityProviders File in utils. I have updated this in next commit